bilby/core/prior/dict.py

```diff
@@ -441,10 +441,26 @@ def constraint_keys(self):
         return [k for k, p in self.items() if isinstance(p, Constraint)]
 
     def sample_subset_constrained(self, keys=iter([]), size=None):
+        efficiency_warning_was_issued = False
+
+        def check_efficiency(n_tested, n_valid):
+            nonlocal efficiency_warning_was_issued
+            if efficiency_warning_was_issued:
+                return
+            efficiency = n_valid / float(n_tested)
+            if n_tested >= 1e3 and efficiency < 1e-3:
+                logger.debug("Warning: prior sampling efficiency is very low, please verify its validity.")
```

Comment:
I think this should be `logger.warning` rather than `debug`, what do you think?

Reply:
Absolutely!